src/Microsoft.Extensions.Configuration.AzureAppConfiguration/JsonConfigurationParser.cs

```diff
@@ -0,0 +1,108 @@
+﻿using Azure.Data.AppConfiguration;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration.AzureKeyVault;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration.FeatureManagement;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Mime;
+using System.Text.Json;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
+{
+    internal static class JsonConfigurationParser
+    {
+        public static bool IsJsonContentType(string contentType)
+        {
+            try
+            {
+                string acceptedMainType = "application";
+                string acceptedSubType = "json";
+
+                ContentType ct = new ContentType(contentType.Trim().ToLower());
+                var type = ct.MediaType;
+                if (Equals(type, FeatureManagementConstants.ContentType)
+                    || Equals(type, KeyVaultConstants.ContentType))
+                {
+                    return false;
+                }
+
+                if (type.Contains('/'))
+                {
+                    string mainType = type.Split('/')[0];
+                    string subType = type.Split('/')[1];
+                    if (Equals(mainType, acceptedMainType))
+                    {
+                        if (subType.Contains('+'))
+                        {
+                            var subTypes = subType.Split('+');
+                            if (Array.Exists(subTypes, x => x == acceptedSubType))
+                            {
+                                return true;
+                            }
+                        }
+                        else if (Equals(subType, acceptedSubType))
+                        {
+                            return true;
+                        }
+                    }
+                }
+            }
+            catch (Exception ex) when (ex is ArgumentException || ex is FormatException || ex is NullReferenceException)
+            {
+                // not a valid JSON content type
+            }
+            return false;
+        }
+
+        public static void ParseJsonSetting(ConfigurationSetting setting, IDictionary<string, ConfigurationSetting> data)
+        {
+            SortedDictionary<string, string> keyValueDict = new SortedDictionary<string, string>();
```

Comment:
>SortedDictionary [](start = 12, length = 16)

Do these settings need to be sorted in a specific order?

Reply:
We don't really need to sort the keys. If the value is a nested JSON object, we could just store keys in the order that they appear inside the value. 
The only thing is, if we don't sort nested objects here, the "data" dictionary will not have sorted keys. When we load settings from config store, we get settings sorted by key. I am not sure if there would be a side effect of having unsorted dictionary of config settings.
Moreover, sorting here will also give the same config output as we would get from reading appsettings.json file (which is not a deciding factor though).

---
In reply to: [453141260](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/181#discussion_r453141260) [](ancestors = 453141260)

Reply:
I would expect the settings to preserve the original ordering from the source instead of being sorted.

---
In reply to: [453933488](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/181#discussion_r453933488) [](ancestors = 453933488,453141260)